app/src/org/odk/collect/android/adapters/HierarchyListAdapter.java

```diff
@@ -41,10 +41,11 @@ public View getView(int position, View convertView, ViewGroup parent) {
             hev = new HierarchyElementView(mContext, mItems.get(position));
         } else {
             hev = (HierarchyElementView)convertView;
-            hev.setPrimaryText(mItems.get(position).getPrimaryText());
-            hev.setSecondaryText(mItems.get(position).getSecondaryText());
-            hev.setIcon(mItems.get(position).getIcon());
-            hev.setColor(mItems.get(position).getColor());
+            HierarchyElement entry = mItems.get(position);
```

Comment:
Might be nice to add a function used by this and the HierarchyElementView that accepts a HierarchyElement and then parse out the colors/texts (since this is a little repeated in HierarchyElementview)